commcare_connect/opportunity/visit_import.py

```diff
@@ -107,7 +107,7 @@ def _bulk_update_visit_status(opportunity: Opportunity, dataset: Dataset):
                 seen_completed_works.add(visit.completed_work_id)
                 status = status_by_visit_id[visit.xform_id]
                 if visit.status != status:
-                    visit.update_status(status)
```

Comment:
I think you need to update [this line](https://github.com/dimagi/commcare-connect/blob/main/commcare_connect/opportunity/visit_import.py#L117) to have the date saved